### PR TITLE
[WIP] Handle `file:/` in Url.parse

### PR DESCRIPTION
**Pull request by kamalaknn**

### CODE REVIEW

File: lib/collection/url.js [hunk 1]
```diff
@@ -355,20 +356,26 @@ _.assign(Url, /** @lends Url */ {
...
 
         // extract the protocol
         p.protocol = _.get(url.match(regexes.extractProtocol), GET_1);
-        _.isString(p.protocol) && (url = url.substr(p.protocol.length + 3)); // remove that damn protocol from url
+        if (_.isString(p.protocol)) {
+            url = url.substr(p.protocol.length); // remove that damn protocol from url
+
+            // remove the scheme host split (://)
+            schemeSeparator = url.match(regexes.schemeSplit);
+            schemeSeparator && (url = url.substr(_.get(schemeSeparator, GET_1).length));
```
Comment (kunagpal) [inline lib/collection/url.js]:
```js
url = url.substr(_.get(schemeSeparator, [GET_1, 'length'], 3));
```

**Comment by kamalaknn** [comment]
Holding this off for now, this slows down parsing time for common case like `https://postman-echo.com/path` and `postman-echo.com/path`

**Comment by kunagpal** [comment]
@kamalaknn Did you mean 'increases parsing time'? 🤔

**Comment by kamalaknn** [comment]
@kunagpal yeah `slows parsing` ~= `increases parsing time`

**Pull request closed without merging**